File: src/main/java/org/dxworks/githubminer/service/repository/commits/GithubCommitService.java
package org.dxworks.githubminer.service.repository.commits;

import com.google.common.reflect.TypeToken;
import org.dxworks.githubminer.dto.response.repository.commits.Commit;
import org.dxworks.githubminer.service.repository.GithubRepositoryService;
import org.dxworks.utils.java.rest.client.providers.BasicAuthenticationProvider;

import java.lang.reflect.Type;
import java.util.Collections;
import java.util.List;

public class GithubCommitService extends GithubRepositoryService {
    private static final Type COMMIT_LIST_TYPE = new TypeToken<List<Commit>>() {
    }.getType();

    public GithubCommitService(String owner, String repo) {
        super(owner, repo);
    }

    public GithubCommitService(String owner, String repo, BasicAuthenticationProvider authenticationProvider) {
        super(owner, repo, authenticationProvider);
    }

    public List<Commit> getAllCommits() {
        return Collections.emptyList();
    }
}

File: src/main/kotlin/org/dxworks/githubminer/service/repository/commits/GithubCommitService.kt
package org.dxworks.githubminer.service.repository.commits

import com.google.api.client.http.GenericUrl
import com.google.common.reflect.TypeToken
import org.dxworks.githubminer.constants.ANONYMOUS
import org.dxworks.githubminer.constants.GITHUB_API_PATH
import org.dxworks.githubminer.dto.response.repository.commits.RepoCommit
import org.dxworks.githubminer.service.repository.GithubRepositoryService

class GithubCommitService(
        owner: String,
        repo: String,
        githubBasePath: String = GITHUB_API_PATH,
        githubTokens: List<String> = listOf(ANONYMOUS)
) : GithubRepositoryService(owner, repo, githubBasePath, githubTokens) {

    val allCommits: List<RepoCommit?>
        get() {
            val commitsUrl = GenericUrl(getApiPath("commits"))
            return paginationUtils.getAllElements(commitsUrl) { it.parseAs(COMMIT_LIST_TYPE) as List<RepoCommit> }
        }

    companion object {
        private val COMMIT_LIST_TYPE = object : TypeToken<List<RepoCommit?>?>() {}.type
    }
}
